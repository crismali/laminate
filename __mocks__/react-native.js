/* eslint-env jest */

class Value {
  constructor (value) {
    this.value = value
  }

  interpolate (args) {
    return { interpolated: this.value, args: args }
  }
}

const AnimatedView = ({ children }) => children
AnimatedView.displayName = 'Animated.View'

const animatedTimingStart = jest.fn()
const Animated = {
  Value,
  View: AnimatedView,
  timing: jest.fn(() => ({ start: animatedTimingStart })),
  animatedTimingStart
}

const Dimensions = { get: jest.fn(() => ({ width: 375, height: 667 })) }

const PanResponder = { create: jest.fn((args) => ({ panResponder: args })) }

const View = ({ children }) => children

const StyleSheet = { create: jest.fn(styles => styles) }

export {
  Animated,
  Dimensions,
  PanResponder,
  StyleSheet,
  View
}
