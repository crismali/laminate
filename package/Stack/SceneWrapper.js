import React, { Component } from 'react'

const { Provider, Consumer } = React.createContext()

export class SceneWrapper extends Component {
  sceneWrapperProps = () => {
    return {
      index: this.props.index,
      interpolateAnimation: this.interpolateAnimation,
      setAnimationValue: this.setAnimationValue
    }
  }

  setAnimationValue = (value) => {
    const { animationValue, index } = this.props
    animationValue.setValue(index + value)
  }

  interpolateAnimation = ({ inputRange, outputRange }) => {
    const { animationValue, index } = this.props
    return animationValue.interpolate({
      inputRange: inputRange.map(value => value + index),
      outputRange
    })
  }

  render () {
    return <Provider value={this.sceneWrapperProps()}>
      {this.props.children}
    </Provider>
  }
}

export const WithStackAnimation = Consumer
