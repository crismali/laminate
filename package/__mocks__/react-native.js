/* eslint-env jest */
import React from 'react'

class Value {
  constructor (value) {
    this.value = value
  }

  interpolate (args) {
    return { interpolated: this.value, args: args }
  }

  setValue (value) {
    this.value = value
  }
}

const AnimatedView = ({ children }) => children
AnimatedView.displayName = 'Animated.View'

const animatedTimingStart = jest.fn((func) => func && func())
const Animated = {
  Value,
  View: AnimatedView,
  timing: jest.fn(() => ({ start: animatedTimingStart })),
  animatedTimingStart
}

const Dimensions = { get: jest.fn(() => ({ width: 375, height: 667 })) }

const Easing = { inOut: jest.fn() }

const PanResponder = { create: jest.fn((args) => ({ panHandlers: args })) }

const View = ({ children }) => <div>{children}</div> // eslint-disable-line

const StyleSheet = { create: jest.fn(styles => styles) }

export {
  Animated,
  Dimensions,
  Easing,
  PanResponder,
  StyleSheet,
  View
}
