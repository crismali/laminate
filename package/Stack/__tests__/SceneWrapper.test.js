/* eslint-env jest */
import React from 'react'
import { Animated } from 'react-native'
import { shallow } from 'enzyme'
import { SceneWrapper } from '../SceneWrapper'

const Child = () => 'a child'

describe('#getChildContext', () => {
  it('has the index, interpolateAnimation, and setAnimationValue', () => {
    const animationValue = new Animated.Value(5)
    const subject = shallow(<SceneWrapper index={17} animationValue={animationValue}>
      <Child />
    </SceneWrapper>)

    expect(subject.instance().getChildContext()).toEqual({
      index: 17,
      interpolateAnimation: subject.instance().interpolateAnimation,
      setAnimationValue: subject.instance().setAnimationValue
    })
  })
})

describe('#setAnimationValue', () => {
  it('sets the animation value to the given value plus the index', () => {
    const animationValue = new Animated.Value(5)
    const subject = shallow(<SceneWrapper index={17} animationValue={animationValue}>
      <Child />
    </SceneWrapper>)
    subject.instance().setAnimationValue(3)
    expect(animationValue.value).toEqual(20)
  })
})

describe('#interpolateAnimation', () => {
  it('interpolates the animation, adjusting the given values by the index', () => {
    const animationValue = new Animated.Value(5)
    const subject = shallow(<SceneWrapper index={17} animationValue={animationValue}>
      <Child />
    </SceneWrapper>)

    const result = subject.instance().interpolateAnimation({ inputRange: [0, 1, 2], outputRange: [5, 2, 4] })
    expect(result).toEqual({
      args: { inputRange: [17, 18, 19], outputRange: [5, 2, 4] },
      interpolated: 5
    })
  })
})

describe('#render', () => {
  it('renders its children', () => {
    const animationValue = new Animated.Value(5)
    const subject = shallow(<SceneWrapper index={3} animationValue={animationValue}>
      <Child />
    </SceneWrapper>)
    expect(subject).toMatchSnapshot()
  })
})
