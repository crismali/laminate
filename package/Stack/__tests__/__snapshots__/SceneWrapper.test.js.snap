// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#render renders its children 1`] = `
<[object Object]
  value={
    Object {
      "index": 3,
      "interpolateAnimation": [Function],
      "setAnimationValue": [Function],
    }
  }
>
  <Child />
</[object Object]>
`;
